test/Microsoft.AspNetCore.Razor.Tools.Test/Infrastructure/ToolTestFixture.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Tools.Test.Infrastructure
+{
+    public class ToolTestFixture : IDisposable
+    {
+        private readonly string _defaultNugetPackageLocation;
+
+        public ToolTestFixture()
+        {
+            PackSrcDirectory();
+
+            _defaultNugetPackageLocation = Environment.GetEnvironmentVariable("NUGET_PACKAGES");
+            Environment.SetEnvironmentVariable("NUGET_PACKAGES", TestArtifactsTempPackagesDirectory);
+
+            InitializeTestApps();
+        }
+
+        public string RootDirectory => Path.GetFullPath(Path.Combine("..", ".."));
```

Comment:
Does this work consistently when running it from VS and commandline? We did a bunch of shenanigans in Mvc to locate the solution root to work around the PWD inconsistencies.